extensions/Worker.Extensions.ServiceBus/src/ServiceBusReceivedMessageConverter.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Core.Amqp;
+using Azure.Messaging.ServiceBus;
+using Microsoft.Azure.Functions.Worker.Converters;
+using Microsoft.Azure.Functions.Worker.Core;
+using Microsoft.Azure.Functions.Worker.Extensions.Abstractions;
+
+namespace Microsoft.Azure.Functions.Worker;
+
+[SupportsDeferredBinding]
+[SupportedConverterType(typeof(ServiceBusReceivedMessage))]
+[SupportedConverterType(typeof(ServiceBusReceivedMessage[]))]
+internal class ServiceBusReceivedMessageConverter : IInputConverter
+{
+    public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+    {
+        ConversionResult result = context?.Source switch
+        {
+            ModelBindingData binding => ConversionResult.Success(ConvertToServiceBusReceivedMessage(binding)),
+            // Only array collections are currently supported, which matches the behavior of the in-proc extension.
+            CollectionModelBindingData collection => ConversionResult.Success(collection.ModelBindingDataArray.Select(ConvertToServiceBusReceivedMessage).ToArray()),
+            _ => ConversionResult.Unhandled()
+        };
+        return new ValueTask<ConversionResult>(result);
+    }
+
+    private ServiceBusReceivedMessage ConvertToServiceBusReceivedMessage(ModelBindingData binding)
+    {
+        // The lock token is a 16 byte GUID
+        const int lockTokenLength = 16;
+
+        if (binding.ContentType != "application/octet-stream")
```

Comment:
We should use [System.Net.Mime](https://learn.microsoft.com/en-us/dotnet/api/system.net.mime.contenttype?view=netcore-2.1) for the stream constant or introduce a string const for this.